TESTS_GUIDELINES.md

```diff
@@ -0,0 +1,54 @@
+# Guidelines for Creating and Executing Tests in Vanilla OS
+
+## Introduction
+
+This document serves as a reference for maintainers of Vanilla OS repositories in creating and managing testing procedures. The goal is to ensure the integrity, security, and functionality of components through consistent and well-structured tests, identifying potential issues before they are brought into production.
+
+It is important that all maintainers are familiar with and have understood the [Vanilla OS contribution guidelines](CONTRIBUTING.md).
+
+## Responsibilities of Maintainers
+
+It is the responsibility of the maintainers to ensure the continuous integration (CI) of tests into their respective projects. Each new feature or fix should be accompanied, where possible, by relevant tests that validate the changes and ensure that existing functionalities remain intact.
+
+## Test Creation Procedure
+
+### Definition of Test Cases
+
+Each test case must strictly correspond to the functional requirements of the component. Clearly define what is being tested and what result is expected.
+
+### Implementation of Tests
+
+For the implementation of tests in projects written in Go, the language predominantly used for backend/critical components of the project, Vanilla OS adopts the official Go testing suite. For other languages, any other suite can be used, considering ease of use and portability, to emphasize its use.
+
+### Automation and Integration of Tests
+
+Integrate the tests into the CI process using GitHub Actions to automate the execution of tests with every push or pull request. This integration ensures that tests are systematically executed, providing immediate feedback on the code's status and preventing the merging of changes that do not pass the tests.
+
+## Quality Standards
+
+### Clean Code
+
+Maintain clean, simple and targeted test code. 
```

Comment:
Not sure if this is the right place, but we should enforce formatters like gofmt (integrated into gopls) for Go and black (or ruff which checks for black's rules) for Python.

Reply:
Not sure if this should be in those guidelines or the general ones

Reply:
Maybe it's best to put in the general guidelines.